### DISCUSSION TITLE
Is there a way to detect when a gridpoint meets a condition and set the time index as the value inplace?

### DISCUSSION THREAD

**Discussion by fridalejandra** [pydata/xarray#6117] [Q&A] [answered]
I have data set with daily sea ice concentration values, I am trying to return one time slice with the same length of lat and lon points which meet the conditions. The condition being: if a cell is =< 15% for three consecutive days, the cell value should be the time index. 

I have managed to write something, but have only gotten it to return the results in the form of a list. 
```
def breakup_date(array):
    final_array = []
    count = 0
    flag = 0
    for j in tq(range(np.shape(array)[1])): #lat
        for k in range(np.shape(array)[2]): #long
            for i in range(np.shape(array)[0] - 1):  #time
                if count == 0:
                    temp_val = array[i,j,k].values
                #print(array[1,j,k].values)
                #print(array[i+1, j,k].values)
                #print()
                if temp_val > 0 and array[i+1,j,k].values <= (temp_val* 0.85):
                    count += 1
                else:
                    count = 0
                if count == 1:
                    timestamp = i
                    lat = j
                    long_ = k

                if count == 3:
                    #print(i,j,k)
                    final_array.append([timestamp,lat,long_])
                 #   print(final_array)
                    flag = 1
                    break
            if flag:
                break
        if flag:
            break
    return np.array(final_array)
```

I would like to deal with this analysis using xarray. I have tried using the rolling window and running a condition on the binary output, but still haven't managed to get just one slice as the output. Is there a 'quicker' way to do this with xarray?

**Comment by max-sixty** [comment, marked as answer]
Those isn't a full answer, but have you tried evaluating the condition at every point, then using a rolling count with a window of 3 to assess where it exists for 3 consecutive points?

**Reply by fridalejandra** [reply]
Tried this and I got it to work! Thanks

**Comment by mathause** [comment]
The [xclim](https://xclim.readthedocs.io/en/stable/)  package may implement some algorithms relevant for your query (you may have to look in the source code for functions related to "at least n consecutive days").

**Reply by fridalejandra** [reply]
Had not heard of this before, thanks for the suggestion!
